subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/codecs/ProviderCodecs.kt

```diff
@@ -207,7 +207,7 @@ ValueSourceProviderCodec(
 class
 PropertyCodec(private val propertyFactory: PropertyFactory, private val providerCodec: FixedValueReplacingProviderCodec) : Codec<DefaultProperty<*>> {
     override suspend fun WriteContext.encode(value: DefaultProperty<*>) {
-        writeClass(value.type!!)
+        writeClass(value.type as Class<*>)
```

Comment:
:x: `DefaultProperty.getType()` should be annotated with `@Nullable` instead of casting here

Reply:
On a first read it seemed like `DefaultProperty.getType()` can't return null. I'll check again.

Reply:
My rationale was that the contract of `Property.getType()` is that its return is `@Nullable` but the override doesn't have the annotation.

Reply:
`DefaultProperty#type` is never null, https://github.com/gradle/gradle/blob/b8327c81db0b082c4561e6ff2b756ac5f7b2f96f/subprojects/core/src/test/groovy/org/gradle/api/internal/model/DefaultObjectFactoryTest.groovy#L48

Reply:
Ah, `@Nullable` is on `Provider.getType()`
And `Property.getType()` doesn't have `@Nullable`.
Then I'm fine with the change, thanks for checking.

Reply:
Confirmed that `DefaultProperty.getType()` can't return null.